code/src/App.js

```diff
@@ -1,9 +1,117 @@
-import React from 'react';
+// /////////////// IMPORT //////////////////////// //
+
+import React, { useEffect, useState } from 'react';
+import { ThoughtsList } from 'Components/ThoughtsList/ThoughtsList';
+import { ThoughtsInputBox } from 'Components/ThoughtInputBox/ThoughtsInputBox';
+import { Footer } from 'Components/Footer/Footer';
+
+// /////////////// MAIN APP //////////////////////// //
+
+// The function starts with declaring,
+// the variables and sets the useState value.
 
 export const App = () => {
+  const [thoughts, setThoughts] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const [newMessage, setNewMessage] = useState('');
+
+  // Here we are calling the API and gets the JSON.
+  // The setLoading (true) shows "loading text" if the API call is delayed.
+  // .catch is catching eventual errors and displays them in a consol.log
+  // No options is created for this request its per default a GET
+
+  const fetchThoughts = () => {
+    setLoading(true)
+    fetch('https://happy-thoughts-ux7hkzgmwa-uc.a.run.app/thoughts')
+      .then((res) => res.json())
+      .then((data) => setThoughts(data))
+      .catch((error) => console.error(error))
+      .finally(() => setLoading(false))
+  }
+
+  // This is an React hook called useEffect. It executes the fetchThoughts function.
+  // This function only runs first time the its mounted.
+
+  useEffect(() => {
+    fetchThoughts(); // <--- This is a callback function. Its executed on every render of the page
+  }, []); // <--- This is the dependency array. When you put something(variables) in here
+  //  it executes every time the array change. But when its empty its only called when mounted.
+  // And ofcourse every time the fetchThouths function is called.
+
+  // This function sets the value of the,
+  // event which is when the user writes a new thought.
+
+  const handleNewThoughtsChange = (event) => {
+    setNewMessage(event.target.value)
+  }
+
+  // This function takes an object as a parameter.
+  // The object in this case is the text the user writes
+  // The event.preventDefault prevents the default behavior for the form
+  // to refresh the page when pressing the submit button.
+
+  const onFormSubmit = (event) => {
```

Comment:
In the onFormSubmit function, you can use object destructuring to extract the newMessage variable instead of accessing it directly:` const { newMessage } = event.target.elements;`